### PR TITLE
Better keyhints

### PR DESCRIPTION
**Pull request by rcorre**

### CODE REVIEW

File: qutebrowser/misc/keyhintwidget.py [hunk 1]
```diff
@@ -90,19 +83,29 @@ def update_keyhint(self, modename, prefix):
...
+            self._show_timer.stop()
             self.hide()
             return
 
+        blacklist = config.get('ui', 'keyhint-blacklist') or []
         keyconf = objreg.get('key-config')
+
+        def blacklisted(keychain):
+            return any([glob for glob in blacklist
+                        if fnmatch.fnmatchcase(keychain, glob)])
```
Comment (The-Compiler) [inline qutebrowser/misc/keyhintwidget.py]: `any()` already checks if all arguments are true, and you can use a [generator expression](https://docs.python.org/3.5/reference/expressions.html#grammar-token-generator_expression).

In other words, you can simplify this to `any(fnmatch.fnmatchcase(keychain, glob) for glob in blacklist)`.

File: qutebrowser/config/configdata.py [hunk 2]
```diff
@@ -355,9 +355,11 @@ def data(readonly=False):
...
              "(requires restart)"),
 
-            ('show-keyhints',
-             SettingValue(typ.Bool(), 'true'),
-             "Show possible keychains based on the current keystring"),
+            ('keyhint-blacklist',
+             SettingValue(typ.List(none_ok=True), ''),
+             "Keychains that shouldn't be shown in the keyhint dialog\n\n"
+             "Globs are supported, so ';*' will blacklist all keychains"
+             "starting with ';'. Use '*' to disable keyhints"),
```
Comment (The-Compiler) [inline qutebrowser/config/configdata.py]: It'd be nice to fix up existing configs with the change - that'd mean adding the option to `RENAMED_OPTIONS` in `config.py` and adding something like `_get_value_transformer({'true': '', 'false': '*'})` to `CHANGED_OPTIONS`.

Comment (rcorre) [inline qutebrowser/config/configdata.py]: I gave it a try switching back and forth to master and it seems to work

File: qutebrowser/misc/keyhintwidget.py (hunk 1 above)
Comment (rcorre) [inline qutebrowser/misc/keyhintwidget.py]: thanks! that's much nicer

**Comment by The-Compiler** [comment]
Thanks!

**Pull request merged**